pkg/app/web/src/components/application-count/index.tsx

```diff
@@ -58,7 +64,7 @@ export const ApplicationCount: FC<ApplicationCountProps> = memo(
           onMouseLeave={handlePopoverClose}
         >
           <Typography variant="h6" component="div" color="textSecondary">
-            {APPLICATION_KIND_TEXT[kind]}
+            {kind}
```

Comment:
as if we keep this `kind` type as ApplicationKind (enum) I think we could make string key name from enum by `ApplicationKind[kind]` (tested with [playground](https://www.typescriptlang.org/play?#code/KYOwrgtgBAggDnANgSwMYEMAuyD2IDSyIAJlAN5QBQUU+AqgEICiASgHJMAqTAylALxQADABpqUbixYwAYgHkWAWQFQAjGJoBhFnJ48ACgBkYHFQCYNUY4oYARGCoDMlzYbl1bLOmxUAWS0yafIIArGIAvpSUqHgAzjiIwAB0iDgA5gAU8EhoWLgERMQA2tkoGNh4hCRJ9MzsXLwAugL8ggBEtawc3DxtAJRAA)) and `APPLICATION_KIND_TEXT` will safe to be removed? 🤔

Reply:
@khanhtc1202 That is a TypeScript feature, but `ApplicationKind` is generated by `protoc` from proto file.
The actual code looks like this.
```js
/**
 * @enum {number}
 */
proto.pipe.model.ApplicationKind = {
  KUBERNETES: 0,
  TERRAFORM: 1,
  CROSSPLANE: 2,
  LAMBDA: 3,
  CLOUDRUN: 4,
  ECS: 5
};
```

So we should define that mapping own 😢.

Reply:
Ah, yes I understand the need of our own defined `ApplicationKind` type (as you mentioned 🙏 ) I just wonder, instead of making this `kind` here as primitive `string` type, keeping it as `ApplicationKind` and at the time when we need to pass a string ApplicationKind name to `Typography`, we could make it by using `ApplicationKind[kind]` as 
```typescript
import { ApplicationKind } from "../../modules/applications";
...
<Typography variant="h6" component="div" color="textSecondary">
  {ApplicationKind[kind]}
</Typography>
```
That what I guess 😄 just to clarify, if we could not do that, then please don't mind of this suggestion 😁

Reply:
👋 I just realize that this `kind` here is a string since we make it as it from here
```typescript
const createInitialCounts = (): Record<string, Record<string, number>> => ({
  [APPLICATION_KIND_TEXT[ApplicationKind.KUBERNETES]]: createInitialCount(),
  [APPLICATION_KIND_TEXT[ApplicationKind.TERRAFORM]]: createInitialCount(),
  [APPLICATION_KIND_TEXT[ApplicationKind.CROSSPLANE]]: createInitialCount(),
  [APPLICATION_KIND_TEXT[ApplicationKind.LAMBDA]]: createInitialCount(),
  [APPLICATION_KIND_TEXT[ApplicationKind.CLOUDRUN]]: createInitialCount(),
  [APPLICATION_KIND_TEXT[ApplicationKind.ECS]]: createInitialCount(),
});

const initialState: ApplicationCounts = {
  updatedAt: 0,
  counts: createInitialCounts(),
};
```
ref [modules/application-counts.ts](https://github.com/pipe-cd/pipe/blob/master/pkg/app/web/src/modules/application-counts.ts#L31-L43)
In that case, rename this `kind` variable as `kindName` (as we do in the caller side [components/application-counts/index.tsx ](https://github.com/pipe-cd/pipe/pull/1730/files/329239fc425c7614d22031d330e0ea7ffba04026#diff-0efde055bddb2f8b7015e7b2cdbbf5f83ad0cb0a9ade472851219cd14bea75e3R55)) may avoid the confusion I caught above 😂 
Sorry for my dumb suggestion 🙏

Reply:
@khanhtc1202 Sorry for my confusing explanation.
```js
proto.pipe.model.ApplicationKind = {
  KUBERNETES: 0,
  TERRAFORM: 1,
  CROSSPLANE: 2,
  LAMBDA: 3,
  CLOUDRUN: 4,
  ECS: 5
};
```
This code is an actual enum code generated by `protoc`.
It just a simple `object`. It not has [reverse mappings](https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings).
So we cannot get a name by using this code `ApplicationKind[ApplicationKind.KUBERNETES]` (This is `undefined`).

Reply:
👀 I see, thank you for clarifying it 🙏